src/lib/services/graphql/graphql-subscriptions.spec.ts

```diff
@@ -0,0 +1,236 @@
+import gql from 'graphql-tag';
+import { WebSocket } from 'mock-socket';
+import { Subscriber } from 'rxjs';
+import { ConnectionStatus } from '../../model/connection-status';
+import { GraphQLSubscriptionsFixture } from './graphql-subscriptions-fixture';
+
+jest.mock('isomorphic-ws', () => WebSocket);
+jest.mock('../../util/sleep-ms', () => ({
+  sleepMs: () => new Promise((resolve) => setTimeout(resolve, 4)),
+}));
+
+/**
+ * The message sequence during these tests is the following:
+ *
+ * client -> server: 'connection_init'
+ * server -> client: 'connection_ack'
+ * client -> server: 'start', ID, query
+ * server -> client: 'data', ID, payload
+ * client -> server: 'end', ID
+ * ...
+ */
+
+// Close codes: https://www.rfc-editor.org/rfc/rfc6455#section-7.4
+describe('GraphQL subscriptions', () => {
+  let fixture: GraphQLSubscriptionsFixture;
+
+  beforeEach(async () => {
+    fixture = new GraphQLSubscriptionsFixture();
+  });
+
+  afterEach(async () => {
+    await fixture.cleanup();
+  });
+
+  it('sends connection init to start', async () => {
```

Comment:
Should we follow the naming convention of test names starting with "should"?

Reply:
This project has 6 tests with should prefix and 255 tests without should prefix